### DISCUSSION TITLE
Structured output with vLLM?

### DISCUSSION THREAD

**Discussion by AkshataDM** [vllm-project/vllm#8300] [General]
Has anyone looked into getting structured output from a model hosted using vllm? Would like some pointers on implementing this if anyone has any suggestions.

**Comment by jeejeelee** [comment]
See: https://github.com/vllm-project/vllm/blob/v0.6.0/vllm/engine/arg_utils.py#L276

**Comment by NicolasDrapier** [comment]
Hi  @AkshataDM 
Yes, I enforce the model to respond with a structured output via `lm-format-enforcer`. In my Docker setup, I add the parameter `--guided-decoding-backend lm-format-enforcer`, and then I call it with the following code:

```python
from pydantic import BaseModel
import json
from openai import OpenAI

class AnswerFormat(BaseModel):
    name: str
    lastname: str
    age: int

client = OpenAI(
    base_url="http://...../v1",
    api_key="token",
)

user_prompt = "......"
completion = client.chat.completions.create(
    model="yourmodelhere",
    messages=[
        {"role": "user", "content": user_prompt}
    ],
    extra_body={
        "guided_json": AnswerFormat.model_json_schema()
    }
)

result = json.loads(completion.choices[0].message.content)
```

**Reply by MohammedAlsayed** [reply]
When I enable --guided-decoding-backend, the function calling options (--enable-auto-tool-choice --tool-call-parser llama3_json --chat-template tool_chat_template_llama3.2_json.jinja)  doesn't execute ... any ideas?

**Comment by datalee** [comment]
must 0.6+ support？
![image](https://github.com/user-attachments/assets/615c5fa7-2ecc-4877-a4c6-cac55ee54c92)
